proto/cosmos/app/v1alpha1/update.proto

```diff
@@ -0,0 +1,26 @@
+syntax = "proto3";
+
+package cosmos.app.v1alpha1;
+
+import "cosmos/app/v1alpha1/config.proto";
+
+// ConfigUpdatePlan describes a planned update to the app config.
+message ConfigUpdatePlan {
+  // height is the block height at which the config update will be performed.
+  int64 height = 1;
+
+  // update are the config updates that will be applied.
+  ConfigUpdates updates = 2;
+}
+
+// ConfigUpdates describes updates to an app config.
+message ConfigUpdates {
+
+  // add_or_update_modules is a list of module configs that will be added if they
+  // describe a new module or updated configs for existing modules with the
+  // same name.
+  repeated ModuleConfig add_or_update_modules = 1;
```

Comment:
I realize this won't actually work. The updated module configs may refer to protobuf types which aren't in the type registry yet because they require some (rolling) binary upgrade first. So the types will be valid when the binaries are ready for the config change but not before.

An alternative is just putting the config as a json string and replacing it entirely. But what about upgrade proposals that happen one after the next? They would just overwrite each other. 

A patching approach like json diff might be better. 

Any ideas?